### DISCUSSION TITLE
After updating Subgen to the latest version, my transcription and translation features are no longer working.

### DISCUSSION THREAD

**Discussion by milesxia** [McCloudS/subgen#138] [Ideas]
Here is the Subgen log file after updating and connecting to Emby, followed by starting to play a movie. Unlike before, it did not start as expected.

subgen  | INFO:     Started server process [27]
subgen  | INFO:     Waiting for application startup.
subgen  | INFO:     Application startup complete.
subgen  | INFO:     Uvicorn running on http://0.0.0.0:9000 (Press CTRL+C to quit)
subgen  | 2024-11-21 08:47:36,804 DEBUG: Raw response: {"Title":"Test Notification","Description":"Test Notification Description","Date":"2024-11-21T08:47:35.1761324Z","Event":"system.notificationtest","Server":{"Name":"c6f740d8a0db","Id":"45387e2ea4e64548ab5f50f1364da246","Version":"4.8.8.0"}}
subgen  | 2024-11-21 08:47:36,815 DEBUG: Emby event detected is: system.notificationtest
subgen  | 2024-11-21 08:47:36,815 INFO: Emby test message received!
subgen  | INFO:     172.29.4.1:54442 - "POST /emby HTTP/1.1" 200 OK
subgen  | 2024-11-21 08:47:43,004 DEBUG: Raw response: {"Title":"admin 在 Chrome macOS 上开始播放 肖申克的救赎","Date":"2024-11-21T08:47:42.9500094Z","Event":"playback.start","User":{"Name":"admin","Id":"dfbc2524b73f40a6b1fb142980279424"},"Item":{"Name":"肖申克的救赎","OriginalTitle":"The Shawshank Redemption","ServerId":"45387e2ea4e64548ab5f50f1364da246","Id":"12","DateCreated":"2024-09-10T09:53:52.0000000Z","Container":"mp4","SortName":"肖申克的救赎","PremiereDate":"1994-09-22T16:00:00.0000000Z","ExternalUrls":[{"Name":"IMDb","Url":"https://www.imdb.com/title/tt0111161"},{"Name":"MovieDb","Url":"https://www.themoviedb.org/movie/278"},{"Name":"TheTVDB","Url":"https://thetvdb.com/dereferrer/movie/190"},{"Name":"Trakt","Url":"https://trakt.tv/search/tmdb/278?id_type=movie"}],"Path":"/media/video/The Shawshank Redemption (1994) [1080p]/The.Shawshank.Redemption.1994.1080p.x264.YIFY.mp4","OfficialRating":"R","Overview":"1947年，小有成就的青年银行家安迪因涉嫌杀害妻子及她的情人而锒铛入狱。在这座名为肖申克的监狱内，希望似乎虚无缥缈，终身监禁的惩罚无疑注定了安迪接下来灰暗绝望的人生。未过多久，安迪尝试接近囚犯中颇有声望的瑞德，请求对方帮自己搞来小锤子。以此为契机，二人逐渐熟络，安迪也仿佛在鱼龙混杂、罪恶横生、黑白混淆的牢狱中找到属于自己的求生之道。他利用自身的专业知识，帮助监狱管理层逃税、洗黑钱，同时凭借与瑞德的交往在犯人中间也渐渐受到礼遇。表面看来，他已如瑞德那样对那堵高墙从憎恨转变为处之泰然，但是对自由的渴望仍促使他朝着心中的希望和目标前进。而关于其罪行的真相，似乎更使这一切朝前推进了一步。","Taglines":["恐惧会囚禁灵魂，而希望能让你自由"],"Genres":["剧情","犯罪"],"CommunityRating":8.707,"RunTimeTicks":85469433330,"Size":1722009470,"FileName":"The.Shawshank.Redemption.1994.1080p.x264.YIFY.mp4","Bitrate":1611813,"ProductionYear":1994,"RemoteTrailers":[{"Url":"https://www.youtube.com/watch?v=PLl99DlL6b4"}],"ProviderIds":{"Tmdb":"278","Imdb":"tt0111161","Tvdb":"190"},"IsFolder":false,"ParentId":"11","Type":"Movie","Studios":[{"Name":"Castle Rock Entertainment","Id":104}],"GenreItems":[{"Name":"剧情","Id":102},{"Name":"犯罪","Id":103}],"TagItems":[],"PrimaryImageAspectRatio":0.6666666666666666,"ImageTags":{"Primary":"c67a406e65a5d410d3dea9e056c664dc","Logo":"8ffe8d46b1ba66e840fafc6170a948c3","Thumb":"dd22d3bda71ff96cf2fd77f2cb1dbfdb"},"BackdropImageTags":["a4bb302f486939774dacad13606840c4"],"MediaType":"Video","Width":1920,"Height":1072},"Server":{"Name":"c6f740d8a0db","Id":"45387e2ea4e64548ab5f50f1364da246","Version":"4.8.8.0"},"Session":{"RemoteEndPoint":"192.168.200.245","Client":"Emby Web","DeviceName":"Chrome macOS","DeviceId":"5a3cd754-03bb-4c08-b58d-03e096171be1","ApplicationVersion":"4.8.8.0","Id":"1a76480c0fec84200cdf1495ec6b2781"},"PlaybackInfo":{"PositionTicks":44380225460,"PlaylistIndex":0,"PlaylistLength":1,"PlaySessionId":"731fd908e80a48f392e13533d8dac241"}}
subgen  | 2024-11-21 08:47:43,004 DEBUG: Emby event detected is: playback.start
subgen  | 2024-11-21 08:47:43,004 DEBUG: Path of file: /media/video/The Shawshank Redemption (1994) [1080p]/The.Shawshank.Redemption.1994.1080p.x264.YIFY.mp4
subgen  | 2024-11-21 08:47:43,005 DEBUG: Updated path: /share/media/video/The Shawshank Redemption (1994) [1080p]/The.Shawshank.Redemption.1994.1080p.x264.YIFY.mp4
subgen  | 2024-11-21 08:47:43,005 DEBUG: /share/media/video/The Shawshank Redemption (1994) [1080p]/The.Shawshank.Redemption.1994.1080p.x264.YIFY.mp4 is an image or is an invalid file or path (are your volumes correct?), skipping processing
subgen  | INFO:     172.29.4.1:54508 - "POST /emby HTTP/1.1" 200 OK

Here is my YAML file used for deployment, with all the environment variables included. I deployed it using QNAP's Container Station through Docker Compose

version: '2'
services:
  subgen:
    container_name: subgen
    tty: true
    image: mccloud/subgen
    environment:
      - "WHISPER_MODEL=small" # 运算大小，越大运算越慢，但是翻译越精准
      - "WHISPER_THREADS=4" # cpu/gpu并发线程数
      - "PROCADDEDMEDIA=True" # 是否在新增媒体时触发工作
      - "PROCMEDIAONPLAY=true" # 是否在媒体播放时触发工作
      - "NAMESUBLANG=chi" # 需要生成的字幕文件语言后缀
      - "SKIPIFINTERNALSUBLANG=chi" # 是否跳过已有中文字幕的视频
      - "SKIPIFEXTERNALSUB=true" # 是否跳过带内嵌字幕的视频
      - "PLEXTOKEN=plextoken" # Plex 媒体服务器的访问令牌
      - "PLEXSERVER=http://plexserver:32400" # Plex 媒体服务器的地址
      - "JELLYFINTOKEN=jellyfintoken" # Jellyfin 媒体服务器的访问令牌
      - "JELLYFINSERVER=http://jellyfinserver:8096" # Jellyfin 媒体服务器的地址
      - "WEBHOOKPORT=9000" # 服务端口
      - "CONCURRENT_TRANSCRIPTIONS=2" # 同时转录的视频数
      - "WORD_LEVEL_HIGHLIGHT=False"  # 高亮显示某单词如同ktv字幕效果
      - "DEBUG=True" # 开启DEBUG日志
      - "USE_PATH_MAPPING=true" # 开启路径重映射
      - "PATH_MAPPING_FROM=/media" # 容器中的媒体库路径
      - "PATH_MAPPING_TO=/share/media" # NAS中的媒体库路径
      - "FORCE_DETECTED_LANGUAGE_TO=zh" # 强制指定转录语言
      - "TRANSCRIBE_OR_TRANSLATE=transcribe" # 一定要设置为 transcribe，仅转录
      - "TRANSCRIBE_DEVICE=cpu" # 使用硬件，有独显可以换成 gpu
      - "CLEAR_VRAM_ON_COMPLETE=True" # 无任务时清除显存
      - "MODEL_PATH=/subgen/models" # 模型保存路径，对应下面的映射
      - "UPDATE=true" # 是否自动从git库更新程序，一些最新更改可能未包含在镜像中
      - "APPEND=False" # 是否在字幕结尾添加 Whisper 文案
      - "USE_MODEL_PROMPT=true" # 强制使用自定义提示词标点符号
      - "CUSTOM_MODEL_PROMPT=你好，欢迎来到我的讲座。" # 自定义提示词
      - "LRC_FOR_AUDIO_FILES=True" # 是否为音频文件生成歌词（LRC 文件）
      - "CUSTOM_REGROUP=cm_sl=84_sl=42++++++1" # 优化字幕格式
      - "SUBGEN_KWARGS={'vad': True,'prompt_reset_on_temperature': 0.35}" # 使用VAD避免幻听，提示前缀避免语言重置
      - "DETECT_LANGUAGE_LENGTH=30" # 检测视频语言的时间长度（秒）
      - "MONITOR=True" # 是否启用文件夹监控
      - "TRANSCRIBE_FOLDERS=/media" # 需要监控并生成字幕的文件夹路径
      - "COMPUTE_TYPE=auto" # 自动选择使用 CPU 或 GPU
      - "RELOAD_SCRIPT_ON_CHANGE=false" # 是否在检测到脚本更改时自动重载
      - "SKIP_LANG_CODES=" # 跳过指定语言的视频字幕生成（为空表示不跳过任何语言）
    volumes:
      - "/share/media:/media" # 将 NAS 的媒体目录挂载到容器内的 /media
      - "/share/Container/subgen/models:/subgen/models" # 将模型文件存储目录挂载到容器
    ports:
      - "9000:9000" # 映射服务端口

**Comment by McCloudS** [comment]
Your log seems to suggest you don’t need to use path mapping, so try disabling. Set USE_PATH_MAPPING to false.

**Reply by milesxia** [reply]
I followed the method you provided and set USE_PATH_MAPPING to false. Below is the log file I obtained. It seems that subgen did not start successfully

VAD:   0% 0/86.66 [00:00<?, ?sec/s]2024-11-22 05:26:58,498 DEBUG: Error processing file /media/.streams/1C/A4/09FD000000000000A511F40000000000/:com.apple.lastuseddate#PS:$DATA
subgen  | 2024-11-22 05:26:58,563 DEBUG: Error processing file /media/.streams/B8/29/09FD000000000000E304E90000000000/:com.apple.lastuseddate#PS:$DATA
subgen  | 2024-11-22 05:26:58,721 DEBUG: Error processing file /media/.streams/00/7E/09FD000000000000F6039F0000000000/:com.apple.lastuseddate#PS:$DATA
subgen  | 2024-11-22 05:26:59,062 DEBUG: Error processing file /media/.streams/44/90/09FD000000000000420B520200000000/:com.apple.lastuseddate#PS:$DATA
subgen  | 2024-11-22 05:26:59,406 DEBUG: Error processing file /media/.streams/D8/26/09FD0000000000007D0FC10000000000/:com.apple.lastuseddate#PS:$DATA
subgen  | 2024-11-22 05:26:59,818 DEBUG: Error processing file /media/.streams/30/9A/09FD000000000000B108C30000000000/:com.apple.lastuseddate#PS:$DATA
subgen  | 2024-11-22 05:26:59,941 DEBUG: Error processing file /media/.streams/E4/2D/09FD0000000000002A09CA0000000000/:com.apple.lastuseddate#PS:$DATA
subgen  | 2024-11-22 05:27:00,094 DEBUG: Error processing file /media/.streams/64/22/09FD00000000000009002A0000000000/:com.apple.provenance:$DATA
subgen  | 2024-11-22 05:27:00,113 DEBUG: Error processing file /media/.streams/64/22/09FD00000000000009002A0000000000/:com.apple.quarantine:$DATA
subgen  | 2024-11-22 05:27:00,151 DEBUG: Error processing file /media/.streams/64/22/09FD00000000000009002A0000000000/:com.apple.metadatakMDItemWhereFroms:$DATA
subgen  | 2024-11-22 05:27:00,678 DEBUG: No subtitles in 'chi' language found in the video.
subgen  | 2024-11-22 05:27:00,701 DEBUG: No subtitles in 'chi' language found in the video.
subgen  | 2024-11-22 05:27:00,757 DEBUG: No subtitles in 'chi' language found in the video.
subgen  | 2024-11-22 05:27:00,805 DEBUG: No subtitles in 'chi' language found in the video.
subgen  | 2024-11-22 05:27:00,835 DEBUG: No subtitles in 'chi' language found in the video.
subgen  | 2024-11-22 05:27:00,863 DEBUG: No subtitles in 'chi' language found in the video.
subgen  | 2024-11-22 05:27:01,068 DEBUG: No subtitles in 'chi' language found in the video.
VAD:   0% 0.16/86.66 [00:03<23:07, 16.05s/sec]2024-11-22 05:27:01,407 DEBUG: No subtitles in 'chi' language found in the video.00<14:31, 10.06s/sec]
VAD:   1% 0.99/86.66 [00:03<02:21,  1.65s/sec]2024-11-22 05:27:01,787 DEBUG: No subtitles in 'chi' language found in the video.00<06:25,  4.45s/sec]
VAD:  12% 10.11/86.66 [00:05<00:24,  3.14sec/s]corrupted double-linked list
subgen  | Traceback (most recent call last):c]
subgen  |   File "/subgen/launcher.py", line 170, in <module>
subgen  |     main()
subgen  |   File "/subgen/launcher.py", line 165, in main
subgen  |     subprocess.run([f'{python_cmd}', '-u', 'subgen.py'], check=True)
subgen  |   File "/usr/lib/python3.10/subprocess.py", line 526, in run
subgen  |     raise CalledProcessError(retcode, process.args,
subgen  | subprocess.CalledProcessError: Command '['python3', '-u', 'subgen.py']' died with <Signals.SIGABRT: 6>.
subgen exited with code 1

**Reply by milesxia** [reply]
Sorry, after restarting the container, it started working. Please open a payment channel—I must find a way to buy you a cup of coffee!

**Reply by McCloudS** [reply]
Glad you got it working!
